lib/Runtime/Library/ByteCodeString.cpp

```diff
@@ -0,0 +1,37 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "RuntimeLibraryPch.h"
+
+using namespace Js;
+
+DEFINE_RECYCLER_TRACKER_PERF_COUNTER(ByteCodeString);
+
+ByteCodeString::ByteCodeString(StaticType * type, const char16* content, charcount_t charLength, Utf8SourceInfo* utf8SourceInfo) :
+    JavascriptString(type, charLength, content),
+    utf8SourceInfo(utf8SourceInfo)
+{
+    AssertMsg(utf8SourceInfo != nullptr, "ByteCodeString requires source info");
+
+#if defined(DBG) && defined(_M_IX86)
+    // Make sure content isn't on the stack by comparing to stack bounds in TIB
+    AssertMsg(!ThreadContext::IsOnStack((void*)content),
+        "ByteCodeString object created using stack buffer...");
+#endif
+
+#if SYSINFO_IMAGE_BASE_AVAILABLE
+    AssertMsg(!AutoSystemInfo::IsJscriptModulePointer((void *)content)
+        && !type->GetScriptContext()->GetRecycler()->IsValidObject((void *)content),
+        "ByteCodeString cannot be used with static or GC strings");
```

Comment:
`ByteCodeString` can't be used with builtin/intl bytecode?

Reply:
Ah, I'd missed that it needs the `Utf8SourceInfo` which we won't have in those cases.

Reply:
For library code we always use LiteralString currently.